DataSourceVMwareGuestInfo.py

```diff
@@ -64,131 +72,306 @@
 #     The metadata key "network.encoding" may be used to indicate the
 #     format of the metadata key "network". Valid encodings are base64
 #     and gzip+base64.
+
+
 class DataSourceVMwareGuestInfo(sources.DataSource):
+
+    dsname = 'VMwareGuestInfo'
+
     def __init__(self, sys_cfg, distro, paths, ud_proc=None):
         sources.DataSource.__init__(self, sys_cfg, distro, paths, ud_proc)
-        self.vmtoolsd = find_executable("vmtoolsd")
-        if not self.vmtoolsd:
+        if not VMTOOLSD:
             LOG.error("Failed to find vmtoolsd")
 
     def get_data(self):
-        if not self.vmtoolsd:
+        """
+        This method should really be _get_data in accordance with the most
+        recent versions of cloud-init. However, because the datasource
+        supports as far back as cloud-init 0.7.9, get_data is still used.
+
+        Because of this the method attempts to do some of the same things
+        that the get_data functions in newer versions of cloud-init do,
+        such as calling persist_instance_data.
+        """
+        if not VMTOOLSD:
             LOG.error("vmtoolsd is required to fetch guestinfo value")
             return False
 
-        # Get the JSON metadata. Can be plain-text, base64, or gzip+base64.
-        metadata = self._get_encoded_guestinfo_data('metadata')
-        if metadata:
-            try:
-                self.metadata = json.loads(metadata)
-            except:
-                self.metadata = safeyaml.load(metadata)
+        # Get the metadata.
+        self.metadata = load_metadata()
+
+        # Apply the network configuration.
+        if not self.apply_network_config():
```

Comment:
Make applying the network configuration directly contingent on a setting in the metadata. This will preserve backwards-compatibility.